spring-integration-redis/src/main/java/org/springframework/integration/redis/util/RedisLockRegistry.java

```diff
@@ -526,6 +588,22 @@ else if (Boolean.FALSE.equals(unlinkResult)) {
 				throw new ConcurrentModificationException("Lock was released in the store due to expiration. " +
 						"The integrity of data protected by this lock may have been compromised.");
 			}
+			else {
+				this.stopRenew();
+			}
+		}
+
+		public final boolean renew() {
+			return Boolean.TRUE.equals(RedisLockRegistry.this.redisTemplate.execute(
+					RENEW_REDIS_SCRIPT, Collections.singletonList(this.lockKey),
+					RedisLockRegistry.this.clientId, String.valueOf(RedisLockRegistry.this.expireAfter)));
+		}
+
+		public final void stopRenew() {
+			if (this.renewFuture != null) {
+				this.renewFuture.cancel(false);
```

Comment:
Why don't use `true` instead?
We definitely are looking for the way to break the currently in the flight request.